network.go

```diff
@@ -150,3 +165,89 @@ func (c *Client) DeleteNetwork(id string) (*SimpleResponse, error) {
 
 	return c.DecodeSimpleResponse(resp)
 }
+
+// GetSubnet gets a subnet with ID
+func (c *Client) GetSubnet(networkID, subnetID string) (*Subnet, error) {
+	resp, err := c.SendGetRequest(fmt.Sprintf("/v2/networks/%s/subnets/", networkID) + subnetID)
+	if err != nil {
+		return nil, decodeError(err)
+	}
+
+	subnet := Subnet{}
+	err = json.NewDecoder(bytes.NewReader(resp)).Decode(&subnet)
+	return &subnet, err
+}
+
+// ListSubnets list all subnets for a private network
+func (c *Client) ListSubnets(networkID string) ([]Subnet, error) {
+	resp, err := c.SendGetRequest(fmt.Sprintf("/v2/networks/%s/subnets", networkID))
+	if err != nil {
+		return nil, decodeError(err)
+	}
+
+	subnets := make([]Subnet, 0)
+	if err := json.NewDecoder(bytes.NewReader(resp)).Decode(&subnets); err != nil {
+		return nil, err
+	}
+
+	return subnets, nil
+}
+
+// CreateSubnet creates a new subnet for a private network
+func (c *Client) CreateSubnet(networkID string, subnet SubnetConfig) (*Subnet, error) {
+	body, err := c.SendPostRequest(fmt.Sprintf("/v2/networks/%s/subnets", networkID), subnet)
+	if err != nil {
+		return nil, decodeError(err)
+	}
+
+	var result = &Subnet{}
+	if err := json.NewDecoder(bytes.NewReader(body)).Decode(result); err != nil {
+		return nil, err
+	}
+
+	return result, nil
+}
+
+// FindSubnet finds a subnet by either part of the ID or part of the name
+func (c *Client) FindSubnet(search, networkID string) (*Subnet, error) {
+	subnets, err := c.ListSubnets(networkID)
+	if err != nil {
+		return nil, decodeError(err)
+	}
+
+	exactMatch := false
+	partialMatchesCount := 0
+	result := Subnet{}
+
+	for _, value := range subnets {
+		if value.Name == search || value.ID == search || value.Label == search {
+			exactMatch = true
+			result = value
+		} else if strings.Contains(value.Name, search) || strings.Contains(value.ID, search) || strings.Contains(value.Label, search) {
+			if !exactMatch {
+				result = value
+				partialMatchesCount++
+			}
+		}
+	}
```

Comment:
We have an endpoint to match the exact name: (`GET /v2/networks/{networkId}/subnets?name=<exact-name>`) so we can use that.
if we want to implement **also** the partial match I think you can proceed the same way without taking care of the exact match (covered by the previous case I just mentioned), collecting all the matching results and returning them like `grep` does (instead of returning an error if there are more matches)